git/gittest/repo.go

```diff
@@ -38,6 +38,10 @@ func Repo(t *testing.T) (git.Repo, func()) {
 		cleanup()
 		t.Fatal(err)
 	}
+	if err = execCommand("git", "-C", filesDir, "tag", "flux-test"); err != nil {
```

Comment:
This will mean that the case of a fresh repo will not be tested. If you want to avoid having to deal with that when testing aspects of the daemon, perhaps add a method here for initialising the tag and use it from there.